src/plugins/vis_type_table/public/components/table_vis_component.tsx

```diff
@@ -20,10 +21,38 @@ interface TableVisComponentProps {
 }
 
 export const TableVisComponent = ({ table, visConfig, handlers }: TableVisComponentProps) => {
-  const { rows, columns } = table;
+  const { formattedRows: rows, formattedColumns: columns } = convertToFormattedData(
+    table,
+    visConfig
+  );
 
   const pagination = usePagination(visConfig, rows.length);
 
+  // set sort and resize column state
+  const [sortColumn, setSortColumn] = useState<SortColumn>(
+    handlers.uiState.get('vis.sortColumn') || {}
+  );
+
+  const [columnsWidth, setColumnsWidth] = useState<ColumnWidth[]>(
+    handlers.uiState.get('vis.columnsWidth') || []
+  );
```

Comment:
Do you actually need these local states? It seems like you can just use `handlers.uiState` getters/setters and the `curColumnsState`.

Reply:
got it. I will combine these local states